File: src/main/java/puzzles/year23/day1/Part1.java
package puzzles.year23.day1;

import util.Puzzle;

public class Part1 implements Puzzle<Integer> {
    @Override
    public Integer getAnswer(String inputString) {
        String[] inputArray = inputString.split("\n");
        int total = 0;

        for (int i = 0; i < inputArray.length; i++) {
            String toTest = inputArray[i];

            char[] toTestArray = toTest.toCharArray();

            String first = "";
            String last = "";



            for (int j = 0; j < toTestArray.length; j++) {
                char c = toTestArray[j];

                if (isNumeric(c)) {
                    if (first.isEmpty()) {
                        first = String.valueOf(c);
                    }

                    last = String.valueOf(c);
                }
            }

            int value = Integer.valueOf(first + last);
            total += value;
        }

        return total;
    }

    public static boolean isNumeric(char c) {
        try {
            Integer.parseInt(String.valueOf(c));

            return true;
        } catch (NumberFormatException e) {
            return false;
        }
    }
}

File: src/main/java/puzzles/year25/day3/Part1.kt
package puzzles.year25.day3

import util.Puzzle

class Part1 : Puzzle<Int?> {
    override fun getAnswer(inputString: String): Int {
        return inputString.lines().sumOf { maxJoltage(it) }
    }

    fun maxJoltage(string: String): Int {
        var max = 0

        for (i in string.toCharArray().indices) {
            for (j in string.substring(i + 1)) {
                val next = string.toCharArray()[i].digitToInt() * 10 + j.digitToInt()
                if (next > max) {
                    max = next
                }
            }
        }

        println(max)

        return max
    }
}
